src/NuGet.Core/NuGet.Packaging/PackageExtraction/ZipArchiveExtensions.cs

```diff
@@ -47,27 +49,41 @@ public static Stream OpenFile(this ZipArchive zipArchive, string path)
             return entry.Open();
         }
 
-        public static string SaveAsFile(this ZipArchiveEntry entry, string fileFullPath)
+        public static string SaveAsFile(this ZipArchiveEntry entry, string fileFullPath, ILogger logger)
         {
             using (var inputStream = entry.Open())
             {
                 inputStream.CopyToFile(fileFullPath);
             }
 
+            entry.UpdateFileTimeFromEntry(fileFullPath, logger);
+
+            return fileFullPath;
+        }
+
+        public static void UpdateFileTimeFromEntry(this ZipArchiveEntry entry, string fileFullPath, ILogger logger)
+        {
             var attr = File.GetAttributes(fileFullPath);
             if (!attr.HasFlag(FileAttributes.Directory) && entry.LastWriteTime.DateTime != DateTime.MinValue)
             {
                 try
                 {
                     File.SetLastWriteTimeUtc(fileFullPath, entry.LastWriteTime.UtcDateTime);
                 }
-                catch (ArgumentOutOfRangeException)
+                catch (ArgumentOutOfRangeException ex)
                 {
-                    // Ignore invalid file times in zip file
+                    if (logger != null)
```

Comment:
done

---
In reply to: [59911236](https://github.com/NuGet/NuGet.Client/pull/504#discussion_r59911236) [](ancestors = 59911236)